integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonAlterTableCompactionCommand.scala

```diff
@@ -276,7 +276,20 @@ case class CarbonAlterTableCompactionCommand(
     val isCompactionTriggerByDDl = true
     val segmentIds: Option[List[String]] = if (compactionType == CompactionType.CUSTOM &&
       alterTableModel.customSegmentIds.isDefined) {
-      alterTableModel.customSegmentIds
+      val ids = alterTableModel.customSegmentIds
+      if (ids != None) {
+        val loadMetadataDetails = carbonLoadModel.getLoadMetadataDetails.asScala
+        for (oneLoad <- loadMetadataDetails) {
```

Comment:
Try best not to use `for` loop in Scala.
You can first filter out the load by matching the id and check whether it is carbon format